### ISSUE TITLE
MessageEvaluator throws a NullReferenceException if the condition argument evaluates to null

### ISSUE THREAD

**Issue by brynrhodes** [DBCG/cql_engine#420]
The MessageEvaluator logic does not account for the possibility that the condition argument will evaluate to null.

**Comment by brynrhodes** [comment]
The following Library exhaustively tests null handling in the Message operator:

```
library TestMessage

// Message(source T, condition Boolean, code String, severity String, message String) T

define TestMessageWithNullSeverity:
  Message(1, true, '1', null as String, 'This is a message')

define TestMessageWithNullSource:
  Message(null, true, '1', 'Message', 'This is a message')

define TestMessageWithNullCondition:
  Message(1, null, '1', 'Message', 'This is a message')

define TestMessageWithNullCode:
  Message(1, true, null, 'Message', 'This is a message')

define TestMessageWithNullMessage:
  Message(1, true, '1', 'Message', null)

define TestWarningWithNullSource:
  Message(null, true, '1', 'Warning', 'This is a message')

define TestWarningWithNullCondition:
  Message(1, null, '1', 'Warning', 'This is a message')

define TestWarningWithNullCode:
  Message(1, true, null, 'Warning', 'This is a message')

define TestWarningWithNullMessage:
  Message(1, true, '1', 'Warning', null)

define TestTraceWithNullSource:
  Message(null, true, '1', 'Trace', 'This is a message')

define TestTraceWithNullCondition:
  Message(1, null, '1', 'Trace', 'This is a message')

define TestTraceWithNullCode:
  Message(1, true, null, 'Trace', 'This is a message')

define TestTraceWithNullMessage:
  Message(1, true, '1', 'Trace', null)

define TestErrorWithNullSource:
  Message(null, true, '1', 'Error', 'This is a message')

define TestErrorWithNullCondition:
  Message(1, null, '1', 'Error', 'This is a message')

define TestErrorWithNullCode:
  Message(1, true, null, 'Error', 'This is a message')

define TestErrorWithNullMessage:
  Message(1, true, '1', 'Error', null)
```

**Comment by JPercival** [comment]
Resolved by #519
